### PR TITLE
Add Owin demo into sln.

### PR DESCRIPTION
**Pull request by ayoung**
There was currently a demo project for `Nancy.Demo.Hostin.Owin` but wasn't sure of its status. (And it was 2 years old!).

This just adds it back in with a demo that works.

### CODE REVIEW

File: src/Nancy.Demo.Hosting.Owin/MainModule.cs [hunk 1]
```diff
@@ -1,98 +1,15 @@
...
-                                   var requestMethod = GetOwinEnvironmentValue<string>(env, "owin.RequestMethod");
-                                   var requestPath = GetOwinEnvironmentValue<string>(env, "owin.RequestPath");
-                                   var owinVersion = GetOwinEnvironmentValue<string>(env, "owin.Version");
-
-                                   return string.Format("You made a {0} request to {1} which runs on owin {2}.", requestMethod, requestPath, owinVersion);
-                               };
-
-            Get["/error1"] = x => View["nope"];
-
+            Get[""] = Root;
```
Comment (horsdal) [inline src/Nancy.Demo.Hosting.Owin/MainModule.cs]: Why cut more or less everything out of the demo. At least some of it I think we should keep.
The `/owin` handler is a good demo IMO. And an `async` handler is good too, I think.

File: src/Nancy.Demo.Hosting.Owin/SimpleOwinAspNetHost.cs [hunk 2]
```diff
@@ -1,623 +0,0 @@
...
-            public const string WebSocketReceiveAsyncFunc = "websocket.ReceiveAsyncFunc";
-            public const string WebSocketCloseAsyncFunc = "websocket.CloseAsyncFunc";
-            public const string WebSocketCallCancelled = "websocket.CallCancelled";
-
-            public const string AspNetWebSocketContext = "System.Web.WebSockets.AspNetWebSocketContext";
-
-            public const string ClientCertificate = "ssl.ClientCertificate";
-        }
-    }
-}
```
Comment (horsdal) [inline src/Nancy.Demo.Hosting.Owin/SimpleOwinAspNetHost.cs]: Nice to see this go, though :clap:

**Comment by ayoung** [comment]
@horsdal how's that? Added the `/owin` demo back in but at the default route instead. Rebased.

**Comment by ayoung** [comment]
Build failed but doesn't seem to be directly from my changes. Please advise.

**Comment by phillip-haydon** [comment]
Restarted the build. See how it goes.

**Comment by phillip-haydon** [comment]
Build successful but branch is out of date.

**Comment by ayoung** [comment]
@phillip-haydon rebased and pushed.

**Comment by khellang** [comment]
This needs another rebase. Sorry about that, @ayoung.

**Comment by horsdal** [comment]
replaced by #2080

**Pull request closed without merging**